### PR TITLE
feat(helm): add global section to helm values

### PR DESCRIPTION
**Pull request by alpiquero**
BREAKING CHANGE: global config has been added to values.yaml:
* `domain` moved to `global.domain`
* `serverName` moved to `global.serverName`
* `tls` moved to `global.ingress.tls.`
* Added `drone.ingress.tls`, `gitea.ingress.tls`, `kdlServer.ingress.tls`, `minio.ingress.tls`, `minio.consoleIngress.tls` and `userToolsOperator.ingress.tls` for individual tls config

### CODE REVIEW

**Comment by jvidal-sysadmin** [comment]
@alpiquero , internal discussion is needed to avoid too many iterations in this improvement.

**Comment by alpiquero** [comment]
Still WIP

**Comment by aacecandev** [comment]
I think we should consider using `http.scheme` in fields that clearly are expressing the protocol to be used.

https://stackoverflow.com/questions/48953298/whats-the-difference-between-a-scheme-and-a-protocol-in-a-url

File: helm/kdl-server/templates/_helpers.tpl [hunk 1]
```diff
@@ -20,12 +20,83 @@ Create MongoDB URI.
...
 */}}
-{{- define "tlsSecretName" -}}
-{{- if .Values.tls.secretName -}}
-  {{- .Values.tls.secretName -}}
+{{- define "global.tlsSecretName" -}}
+{{-  if hasKey .Values.global.ingress.tls "secretName" -}}
+  {{- .Values.global.ingress.tls.secretName -}}
 {{- else -}}
-  {{- printf "%s-tls" $.Values.domain -}}
+  {{- printf "%s-%s-tls" $.Values.global.domain $.appName -}}
```
Comment (blopezpi) [inline helm/kdl-server/templates/_helpers.tpl:29]: I would like to include a replace here to replace "." with "-" `| replace "." "-"` .

File: helm/kdl-server/templates/_helpers.tpl [hunk 2]
```diff
@@ -20,12 +20,83 @@ Create MongoDB URI.
...
 
 {{/*
-Create tls secret name
+Global tls secret name
 */}}
-{{- define "tlsSecretName" -}}
-{{- if .Values.tls.secretName -}}
-  {{- .Values.tls.secretName -}}
+{{- define "global.tlsSecretName" -}}
+{{-  if hasKey .Values.global.ingress.tls "secretName" -}}
```
Comment (aacecandev) [inline helm/kdl-server/templates/_helpers.tpl:26]: There's an unnecessary extra space here

File: helm/kdl-server/templates/gitea/init-configmap.yaml [hunk 3]
```diff
@@ -21,15 +21,15 @@ data:
...
-    ROOT_URL         = http://gitea.{{ .Values.domain }}/
+    ROOT_URL         = http://gitea.{{ .Values.global.domain }}/
     DISABLE_SSH      = false
     SSH_PORT         = 22
     SSH_LISTEN_PORT  = 22
     LFS_START_SERVER = true
     LFS_CONTENT_PATH = /data/git/lfs
-    DOMAIN           = gitea.{{ .Values.domain }}
+    DOMAIN           = gitea.{{ .Values.global.domain }}
     LFS_JWT_SECRET   = dJd7zxw11VRhqCighcaeLvUbxlDC45IK2VNc4TCsZc8
```
Comment (aacecandev) [inline helm/kdl-server/templates/gitea/init-configmap.yaml:33]: This

File: helm/kdl-server/values.yaml [hunk 4]
```diff
@@ -72,6 +83,9 @@ drone:
   adminToken: 7GSipOV0wJZQioZNBxaw3AotHV1tA4K4
```
Comment (aacecandev) [inline helm/kdl-server/values.yaml:83]: This

File: scripts/kdlctl/cmd_deploy.sh [hunk 5]
```diff
@@ -111,23 +111,28 @@ deploy_helm_chart() {
     --set backup.image.tag="latest" \
     --set domain="${DOMAIN}" \
     --set drone.storage.storageClassName="${STORAGE_CLASS_NAME}" \
+    --set drone.ingress.tls.secretName="$DOMAIN-tls-secret" \
```
Comment (aacecandev) [inline scripts/kdlctl/cmd_deploy.sh:114]: We should not use DOMAIN as a variable name here, it is super confusing, especially after removing `.Values.domain` and making Helm fail when the value is present

**Review by jvidal-sysadmin — APPROVED** [review]

File: helm/kdl-server/templates/gitea/init-configmap.yaml (hunk 3 above)
Comment (aacecandev) [reply helm/kdl-server/templates/gitea/init-configmap.yaml:33]: this

**Review by aacecandev — APPROVED** [review]

File: helm/kdl-server/templates/_helpers.tpl (hunk 1 above)
Comment (blopezpi) [reply helm/kdl-server/templates/_helpers.tpl:29]: @alpiquero could you have an opportunity to see that?

**Pull request merged**